### ISSUE TITLE
setGroup() not working for new players

### ISSUE THREAD

**Issue by suresttexas00** [benbaptist/minecraft-wrapper#238]
When a new player joins, my plugins cannot use `playerobject.setGroup("groupname")` to add the group.  No error is generated; the group is simply never added.

However, if I type (as OP) the command `/perms users <player> info` first and then run the setGroup command (via plugin code/command), the operation succeeds and the player is added to the group.

Something being done by the info subcommand of __commands.py__ that allows the setGroup() command in __api/player.py__ to function as intended.

The code from each section...
```
""" api/player
	def setGroup(self, group):
		"" Adds the player to a specified group. If the group does not exist, an IndexError is raised. ""
		if not group in self.permissions["groups"]:
			raise IndexError("No group with the name '%s' exists" % group)
**		for uuid in self.permissions["users"]:
			if uuid == str(self.uuid):
				self.permissions["users"][uuid]["groups"].append(group)
"""
value = self.wrapper.permissions["users"][uuid]["permissions"][node]

""" Commands
					elif subcommand == "info":
						player.message("&aUser '%s' is in these groups: " % username)
						for group in self.wrapper.permissions["users"][uuid]["groups"]:
							player.message("- %s" % group)
						player.message("&aUser '%s' is granted these individual permissions (not including permissions inherited from groups): " % username)
**						for node in self.wrapper.permissions["users"][uuid]["permissions"]:
							value = self.wrapper.permissions["users"][uuid]["permissions"][node]
							if value == True:
								player.message("- %s: &2%s" % (node, value))
							elif value == False:
								player.message("- %s: &4%s" % (node, value))
							else:
								player.message("- %s: &7%s" % (node, value))
"""
```

I can only suspect that the statement `if uuid == str(self.uuid):` perhaps is not parsing as True?

**Comment by benbaptist** [comment]
Ah, interesting. I need to centralize this code so badly into a more organized chunk, to guarantee that the file gets generated properly.

**Comment by suresttexas00** [comment]
I found what fixes this.  

the player needs a setUUID for setGroup to work.  This is accomplished by adding a statement to setGroup and hasGroup():

```self.uuid = self.wrapper.lookupUsername(self.username)```

lookupUsername also needs fixed to ensure the properly capped name is sent to setUUID(): 
```
		try:
			r = requests.get("https://api.mojang.com/users/profiles/minecraft/%s" % username)
			uuid = self.formatUUID(r.json()["id"])
			correctcapname = self.formatUUID(r.json()["name"])
		except:
			uuid=self.wrapper.getUUID(username)
			correctcapname = username
		self.setUUID(uuid, correctcapname)
		return uuid
```

... part of the on-going fixes that are part of PR #288

**Comment by suresttexas00** [comment]
#292
